### PR TITLE
Always Scale :class:`~.Text` according to the screen size

### PR DESCRIPTION
**Pull request by naveen521kk**
## Changelog / Overview
This should help people because they don't need to scale
every time instead it will automatically scale.

It's an initial implementation, only for `Text` now. It
can be extended to other implementations also.

This also fixes https://github.com/ManimCommunity/manim/issues/1327
by adding an option to disable word wraps.

:class:`Text` now scales itself according to Video Size or Screen Size it renders on.
This would generate videos differently as compared to the previous versions.

Also, a new option called :attr:`wrap_text` is added to :class:`Text`, which can
be used to disable word wraps, and is enabled by default.

## Motivation
Improves the `Text` object. Also, if this comes good, the same would be added to `MarkupText` also.

## Explanation for Changes
Previously, this script produced this output.
```py
from manim import *

lorem_ipsum = """Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."""


class Test(Scene):
    def construct(self):
        a = Text(lorem_ipsum[:60])
        self.play(FadeIn(a))
```
test-scene-manim

With these changes, it produces this output.

test-image-updated

## Testing Status
Works locally, on Windows.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by naveen521kk** [comment]
I will make these change optional by a flag called `scale_automatically`.

**Comment by cobordism** [comment]
My feeling is that this is overkill and also partially unwanted.

Let me explain:
It does not _have_ to be a global config... but it could be. 
It could also just be a keyword argument to Text. We already have so many keyword arguments for various options that are not all accessible through global config, why should this one be? It seems a lot of overhead when I could just do: `def myText(*args, **kwargs): Text(*args, scale_to_fit_screen="always", **kwargs)`

Secondly, I really don't like the idea of automatically scaling text to be LARGER in order to fit the screen.  This seems unwanted behaviour.


In my view, a minimally invasive solution would be the following:

- add a keyword argument to Text `scale_to_screen` that takes values `always`, `never`, and `auto` with default `auto`.
- In auto mode, when text is encountered that would overflow the page, scale it down to fit the screen _and_ print a warning message that the text was scaled down.

- optionally make the default settable in the global config.

**Comment by naveen521kk** [comment]
>  It does not have to be a global config... but it could be.

The main reason I did so, was that examples would have a lot of the same keyword argument which may confuse users. 

> It could also just be a keyword argument to Text.

It already is, `scale_automatically` it is called and accepts either `True` or `False`. If `True`(by default), it is scaled to fit the screen, while if it `False` the older method is used.

> Secondly, I really don't like the idea of automatically scaling text to be LARGER in order to fit the screen. This seems unwanted behaviour.

Yeah. I think one way for this is that our SVG parser should look for the size specified in the SVG file itself. For now, both `Text` and `Tex` object just scale with an arbitrary constant. I need to look into what can be done here.

**Comment by cobordism** [comment]
>  If True(by default), it is scaled to fit the screen, while if it False the older method is used.

Yes, and I think it is really unhelpful if a small text is suddenly scaled to fit the screen. 
I understand why you might want to automatically scale down if otherwise it does not fit the screen, but not to scale up single words or phrases

File: manim/mobject/svg/text_mobject.py [hunk 1]
```diff
@@ -11,14 +11,15 @@
 
     class HelloWorld(Scene):
         def construct(self):
-            text = Text('Hello world').scale(3)
+            text = Text('Hello world')
```
Comment (cobordism) [inline manim/mobject/svg/text_mobject.py:14]: are you saying that you automatically scale this up to fit the screen?
What happens if I keep the `scale(3)`?

Comment (naveen521kk) [reply manim/mobject/svg/text_mobject.py:14]: scale-3
^ when scale=3

when there is no scale it generates

HelloWorld_ManimCE_v0 5 0

**Comment by naveen521kk** [comment]
I still think this can be improved still about not enlarging text to fit the screen. Until it is ready I am having is as a draft.

File: manim/mobject/svg/text_mobject.py (hunk 1 above)
Comment (cobordism) [reply manim/mobject/svg/text_mobject.py:14]: ah. well may I ask: why did you remove the scale(3) from the example?

**Comment by cobordism** [comment]
The issue I have with auto-enlarging is that it will be near impossible to have small text snippets be of the same size. If I do
```
t1 = Text("Hi")
t2 = Text("Hello")
t3 = Text("How do you do")
```
will the first be huge, the second large, and the third normal? And how much would I need to scale them to make them the same font size?

That is why I think we should only offer scaling down to fit the screen automatically.

**Comment by naveen521kk** [comment]
It would be rather easy to implement this after https://github.com/ManimCommunity/manim/issues/1390 so closing till then.

**Pull request closed without merging**